orchagent/portsorch.cpp

```diff
@@ -1860,6 +1885,33 @@ void PortsOrch::initPortSupportedSpeeds(const std::string& alias, sai_object_id_
     m_portStateTable.set(alias, v);
 }
 
+void PortsOrch::initPortCapLinkTraining(Port &port)
+{
+    sai_status_t status;
+    sai_attribute_t attr;
+
+#ifdef SAI_PORT_ATTR_SUPPORTED_LINK_TRAINING_MODE
+    attr.id = SAI_PORT_ATTR_SUPPORTED_LINK_TRAINING_MODE;
+#else
+    /*
+     * Fallback to autoneg upon legacy SAI implementation
+     * In the case of SFP/QSFP ports, link-training is most likely available
+     * if autoneg is supported.
+     */
+    attr.id = SAI_PORT_ATTR_SUPPORTED_AUTO_NEG_MODE;
+#endif
+    status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        port.m_cap_lt = attr.value.booldata ? 1 : 0;
+    }
+    else
+    {
+        port.m_cap_lt = 1; /* Default to 1 for vstest */
```

Comment:
I am not sure about this. It means that m_cap_lt will be 1 even if SAI return error. I would suggest to implement unit test if VS test does not support link training.

Reply:
This is an issue in VS, do you know where is the source of this SAI model? SAI/bm/p4-sai?
Could we this VS enhancement deal in a separate PR? We'll need to do this for both AutoNeg and LinkTraining